Sign button change on certificate upload

brumik opened the pull request:
https://issues.redhat.com/browse/AAH-1369 - update

When the flag `can_upload_signatures` is set it takes precedence over the backend signing. Therefore we are hiding the `Sign all` buttons on the collection and namespace detail pages, and changing the `Sign version x.x` to bring up the certificate upload modal. 

In addition it fixes the `waitForTask` function to return error if the task is not `running` and not `completed` so we can report failures.

**After with the `can_upload_signatures = true`**:
![Screenshot from 2022-06-20 13-06-02](https://user-images.githubusercontent.com/8531681/174588976-44754a7b-947b-4df8-a9c5-af3351a110ac.png)
![Screenshot from 2022-06-20 13-06-42](https://user-images.githubusercontent.com/8531681/174588979-35235fd7-d5c0-48ab-bfd8-93ec72f7dfb7.png)
![Screenshot from 2022-06-20 13-07-09](https://user-images.githubusercontent.com/8531681/174588982-f1d8af35-0fa2-4240-8466-fe7c5d4cef68.png)
![Screenshot from 2022-06-20 13-07-15](https://user-images.githubusercontent.com/8531681/174588986-37885d6f-6ba7-401f-b9b0-ccddee7cd7f9.png)
![Screenshot from 2022-06-20 13-07-31](https://user-images.githubusercontent.com/8531681/174588988-23165c03-b05b-4f35-9897-c469133d681d.png)
![Screenshot from 2022-06-20 13-08-41](https://user-images.githubusercontent.com/8531681/174588990-aba9c7f4-98ac-4c98-aa57-cf5f438dc8d6.png)

ShaiahWren committed:
Edit remote repo (#2064)

* edit remote repo

Issue: AAH-618

* edit a remote repo

Issue: AAH-618

* adjust tests for rh-certified

Issue: AAH-618

* add to changelog

Issue: AAH-618

* check for visibility on cy.eq() calls

Issue: AAH-618

* remove config button test - it won't appear anyway

No-Issue

* confirm all values are cleared

Issue: AAH-618

drodowic committed:
enable ephemeral pr_check.sh (#2087)

* add ephemeral pr and build scripts

Issue: AAH-1598

ShaiahWren committed:
Only show `add ee` button with user permissions (#2209)

* only show add EE button with user model permissios

Issue: AAH-1696

* remove div for spacing if add button is missing.

Issue: AAH-1696

ShaiahWren committed:
show sync button only with user permissions for changing registry (#2190)

* show sync button only with user permissions for changing registry

Issue: AAH-1695

* include model permissions logic in buttons variable

Issue: AAH-1695

ShaiahWren committed:
show sync button in toggle only with correct user permissions (#2189)

* how sync button only with user permissions for changing container

Issue: AAH-1694

* use canEdit to determind sync buton visibility

Issue: AAH-1694

Commit:
Automated updated of i18n strings on 2022-06-19

dependabot[bot] committed:
Bump @typescript-eslint/eslint-plugin from 5.27.1 to 5.28.0 (#2216)

Bumps [@typescript-eslint/eslint-plugin](https://github.com/typescript-eslint/typescript-eslint/tree/HEAD/packages/eslint-plugin) from 5.27.1 to 5.28.0.
- [Release notes](https://github.com/typescript-eslint/typescript-eslint/releases)
- [Changelog](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/CHANGELOG.md)
- [Commits](https://github.com/typescript-eslint/typescript-eslint/commits/v5.28.0/packages/eslint-plugin)

---
updated-dependencies:
- dependency-name: "@typescript-eslint/eslint-plugin"
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump @babel/core from 7.18.2 to 7.18.5 (#2215)

Bumps [@babel/core](https://github.com/babel/babel/tree/HEAD/packages/babel-core) from 7.18.2 to 7.18.5.
- [Release notes](https://github.com/babel/babel/releases)
- [Changelog](https://github.com/babel/babel/blob/main/CHANGELOG.md)
- [Commits](https://github.com/babel/babel/commits/v7.18.5/packages/babel-core)

---
updated-dependencies:
- dependency-name: "@babel/core"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump webpack from 5.72.1 to 5.73.0 (#2214)

Bumps [webpack](https://github.com/webpack/webpack) from 5.72.1 to 5.73.0.
- [Release notes](https://github.com/webpack/webpack/releases)
- [Commits](https://github.com/webpack/webpack/compare/v5.72.1...v5.73.0)

---
updated-dependencies:
- dependency-name: webpack
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump eslint from 8.16.0 to 8.18.0 (#2217)

Bumps [eslint](https://github.com/eslint/eslint) from 8.16.0 to 8.18.0.
- [Release notes](https://github.com/eslint/eslint/releases)
- [Changelog](https://github.com/eslint/eslint/blob/main/CHANGELOG.md)
- [Commits](https://github.com/eslint/eslint/compare/v8.16.0...v8.18.0)

---
updated-dependencies:
- dependency-name: eslint
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

brumik committed:
Change sign buttons when certificate upload is enabled

Issue: AAH-1369

brumik commented:
@himdel I will rebase/reopen the PR against ansible when your PR gets merged.

dependabot[bot] committed:
Bump react-router-dom from 5.3.2 to 5.3.3 (#2213)

Bumps [react-router-dom](https://github.com/remix-run/react-router/tree/HEAD/packages/react-router-dom) from 5.3.2 to 5.3.3.
- [Release notes](https://github.com/remix-run/react-router/releases)
- [Commits](https://github.com/remix-run/react-router/commits/v5.3.3/packages/react-router-dom)

---
updated-dependencies:
- dependency-name: react-router-dom
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

himdel commented on src/components/headers/collection-header.tsx:
```diff
@@ -639,6 +662,73 @@ export class CollectionHeader extends React.Component<IProps, IState> {
+      .then((result) => {
+        // This is a hack because it returns the full task api path:
+        // eg.: /api/automation-hub/pulp/api/v3/tasks/0be64cb4-3b7e-4a6b-b35d-c3b589923a90/
+        return waitForTask(result.data.task.slice(0, -1).split('/').pop()).then(
```
```suggestion
        return waitForTask(parsePulpIDFromURL(result.data.task)).then(
```

(+ `import { parsePulpIDFromURL } from 'src/utilities';`)

himdel commented:
:+1: looks good :)

ShaiahWren committed:
Add loading page to all pages (#2212)

* add LoadingPageWithHeader to token management

Issue: AAH-518

* add loadingToken state object

Issue: AAH-518

* use LoadingPageSpinner for imports list

Isssue: AAH-518

* changelog entry

Issue: AAH-518

* add LoadingPageSpinner while loading Token

Issue: AAH-518

* remove loading variable, only use loadingToken

Issue: AAH-518

himdel committed:
Update collection signing feature flags (#2211)

* Update collection signing feature flags

Issue: AAH-1710

* use new signing feature flags

* remove signing_enabled where other featureFlags are also present

* sign state - add display_signatures check around Signed and approved

* Sign and approve only unless require_upload_signatures

on approval dashboard prioritize `require_upload_signatures`:
if true, show `[upload | approve]` buttons
if false, then you can show `[sign and approve]` if `collection_auto_sign` is true

if `collection_auto_sign` is false and `require_upload_signatures` is false, show only `[approve]`
